### PR TITLE
Fixes #251: WP Retina 2x compatibility

### LINKED ISSUE #251: WP Retina 2x
Fix the whole thing.
[Source](https://app.intercom.io/a/apps/cd6nxj3z/inbox/inbox/1295841/conversations/15590918148).

### PR DESCRIPTION
**Pull request by Screenfeed**

### CODE REVIEW

**Comment by Tabrisrp** [comment]
you have a conflict to fix too

File: inc/3rd-party/wp-retina-2x/inc/classes/class-imagify-wp-retina-2x.php [hunk 1]
```diff
@@ -0,0 +1,683 @@
...
+	 * @return string Path to the temporary file.
+	 */
+	public function get_uploaded_file_path() {
+		global $wr2x_core;
+
+		$tmp_file_path = ! empty( $_FILES['file']['tmp_name'] ) && is_uploaded_file( $_FILES['file']['tmp_name'] ) ? $_FILES['file']['tmp_name'] : '';
+		$filesystem    = Imagify_Filesystem::get_instance();
+
+		if ( ! $tmp_file_path || ! $filesystem->is_image( $tmp_file_path ) ) {
+			$wr2x_core->log( 'The file is not an image or the upload went wrong.' );
```
[the remark this answers is not in the document]
Comment (Screenfeed) [reply inc/3rd-party/wp-retina-2x/inc/classes/class-imagify-wp-retina-2x.php]: OK, I'll do more tests.

File: inc/3rd-party/wp-retina-2x/inc/classes/class-imagify-wp-retina-2x-core.php [hunk 2]
```diff
@@ -0,0 +1,1539 @@
...
+
+		$original_dirpath = $this->filesystem->dir_path( $attachment->get_original_path() );
+
+		foreach ( $sizes as $size_key => $image_data ) {
+			$retina_path = wr2x_get_retina( $original_dirpath . $image_data['file'] );
+
+			if ( $retina_path ) {
+				// The file exists.
+				$sizes[ $size_key ]['retina-path'] = $retina_path;
+			} else {
```
[the remark this answers is not in the document]
Comment (Screenfeed) [reply inc/3rd-party/wp-retina-2x/inc/classes/class-imagify-wp-retina-2x-core.php]: Mouais... For one line of code...

File: inc/3rd-party/wp-retina-2x/inc/classes/class-imagify-wp-retina-2x-core.php [hunk 3]
```diff
@@ -0,0 +1,1539 @@
...
+		 * @param int   $attachment_id      The attachment ID.
+		 * @param array $sizes              An array of non-retina thumbnail sizes.
+		 * @param int   $optimization_level The optimization level.
+		 */
+		do_action( 'before_imagify_optimize_retina_images', $attachment_id, $sizes, $optimization_level );
+
+		$imagify_data = $attachment->get_data();
+
+		foreach ( $sizes as $size_key => $image_data ) {
+			$imagify_data = $this->optimize_retina_image( array(
```
[the remark this answers is not in the document]
Comment (Screenfeed) [reply inc/3rd-party/wp-retina-2x/inc/classes/class-imagify-wp-retina-2x-core.php]: Yep, that's the idea (this method adds data to it).

**Pull request merged**